### ISSUE TITLE
integration into the official repo?

### ISSUE THREAD

**Issue by oconnor663** [sneves/blake2-avx2#3]
Are there any plans to incorporate this version into the [official BLAKE2 implementation](https://github.com/BLAKE2/BLAKE2)?

I've noticed that on my system (Arch Linux, Intel i5-4590), `sha1sum` is substantially faster than `b2sum`, even when `b2sum` is compiled directly from that repo with `-march=native`. I think that's because `sha1sum` is linking against OpenSSL's SHA1 implementation, which uses AVX2. Is that right?

**Comment by sneves** [comment]
I agree that this should happen at some point, I've just been occupied with other things..

Ideally getting all these fast implementations into OpenSSL would be the way to go, would make maintaining a separate library with its own ISA detection unnecessary.

**Comment by jedisct1** [comment]
libsodium has this code since... well... pretty much since Samuel wrote it :)
